tests/tarp_test.py

```diff
@@ -141,12 +141,46 @@ def test_run_tarp_correct(distance, z_score_theta, accurate_samples):
         num_bins=30,
     )
 
-    assert allclose((ecp - alpha).abs().max(), Tensor([0.0]), atol=1e-1)
+    assert allclose((ecp - alpha).abs().max(), zeros((1,)), atol=1e-1)
     assert (
         ecp - alpha
     ).abs().sum() < 1.0  # integral of residuals should vanish, fig.2 in paper
 
 
+@pytest.mark.gpu
+@pytest.mark.parametrize("distance", (l1, l2))
+@pytest.mark.parametrize("z_score_theta", (True, False))
+def test_run_tarp_correct_on_cuda_device(distance, z_score_theta, accurate_samples):
+    dev = device("cuda")
+    theta, samples = accurate_samples
+    theta, samples = theta.to(dev), samples.to(dev)
+
+    with pytest.raises(NotImplementedError):
+        # let's make sure the execution problem is still there
+        # if torch fixes https://github.com/pytorch/pytorch/issues/69519
+        # this context manager should ensure, the case fails
+        # then we can fix the tarp code
+        from torch import histogram
+
+        histogram(zeros((3,)).cuda(), bins=4)
```

Comment:
I am not sure I understand why need this here. To be sure: 
below we are calling `_run_tarp` which uses `histogram` to calculate the `ecp` values. it will move the coverage values to `cpu` to do so. 
here we are explicitly calling `histogram` with values on the gpu and catching the error? and the purpose is that we are notified when this is not failing anymore?

Reply:
correct! This is code to notify us once/if pytorch fixes this problem and computing the histogram can be done on the GPU. Otherwise, the line in the tarp code which moves data to the CPU just because torch doesn't support anything else might stay and not get fixed.

Reply:
@janfb just wanted to ping you on this as it is marked as unresolved.